apps/frontend/src/app/Types/character.d.ts

```diff
@@ -65,4 +64,13 @@ export type ICharTC = {
     },
     sets: Partial<Record<ArtifactSetKey, 1 | 2 | 4>>
   }
+  optimization: {
+    target?: string[]
+    distributedSubstats: number
+    maxSubstats: Record<SubstatKey, number> & {

```

Comment:
why not simplify it to just Record<SubstatKey, number>? 
I know you designed this with the globalmax+offset in mind, but I want to gear this UI to be usable for general users as well, not just KQMS. 
I rather the user do a bit more repetitive clicks to get their results, rather than the UI to be unintuitive to the point where they can't use it.

Reply:
This is just the storage for the UI bits. If you want I can entirely remove the globalmax+offset from the UI, but that would make it slower to use for KQMS

Reply:
for UI:
- have "allocated subs"(like what you have now, it ticks down to zero when optimizer assign subs)
- allow user to enter a "max" for each substat (i think you have those red inputs there)
- no offset (confusing)

its a bit more work on the user side to define each max, but I assume only ppl for KQMS will define maxes, so its not too bad (could always add a KQMS button as shortcut later, if they request, but I want to cater this UI to general users)

general user will just put in a reasonable number for allocated subs and click "optimize"

and eventually ill make it so that you can't put decimal substats, and make everything in this UI work in units of subs

Reply:
Also, need to feed all the combination of substats through the `uiDataForTeam` API for correctness, and in theory all the calculation funkiness should go away.